packages/frontend/src/components/chart/configure/render/render.ts

```diff
@@ -0,0 +1,81 @@
+import { toDays } from '../controls/toDays'
+import { ChartElements } from '../elements'
+import { State } from '../state/State'
+import { renderChart } from './renderChart'
+import { renderHover } from './renderHover'
+
+export function render(
+  elements: ChartElements,
+  previousState: State,
+  state: State,
+) {
+  if (previousState === state) {
```

Comment:
Is ref comparison here intentional?

Reply:
Yes. There is a super rare edge case where multiple updates happen in the same animation frame and the previous render has already drawn the latest state, so we can skip the work. Potentially we can just remove this check

Reply:
Sounds good, let's keep it then.

Reply:
I've added a comment explaining it too :)